Lib/ipaddress.py

```diff
@@ -594,6 +594,76 @@ def _get_address_key(self):
     def __reduce__(self):
         return self.__class__, (self._ip,)
 
+    def __format__(self, fmt):
+        """Returns a formatted string. Supported presentation types are:
+            's': returns the IP address as a string (default)
+            'b' or 'n': converts to binary and returns a zero-padded string
+            'X' or 'x': converts to upper- or lower-case hex and returns a zero-padded string
+
+            For binary and hex presentation types, the alternate form specifier
+            '#' and the grouping option '_' are supported.
+
+        """
+
+
+        # string
+        if len(fmt) == 0 or fmt[-1] == 's':
+            # let format() handle it
+            return format(str(self), fmt)
+
+        # from here on down, support for 'bnXx'
+
+        import re
+        fmt_re = re.compile('^(#?)(_?)(x|b|n|X){1}$')
```

Comment:
You can use `([xbnX])` instead of the alternation.  It might also be clearer to use named groupings.

Reply:
Done.